Classes/ARLocalTiledImageDataSource.m

```diff
@@ -11,16 +11,51 @@
 
 @implementation ARLocalTiledImageDataSource
 
+- (id)init
+{
+    self = [super init];
+    if (self) {
+        _extralVisualLevel = 1;
+    }
+    return self;
+}
+
 - (UIImage *)tiledImageView:(ARTiledImageView *)imageView imageTileForLevel:(NSInteger)level x:(NSInteger)x y:(NSInteger)y
 {
-    NSString *filename = [NSString stringWithFormat:@"%@/%@/%@_%@.%@", self.tileBasePath, @(level), @(x), @(y), self.tileFormat];
-    return [UIImage imageWithContentsOfFile:filename];
+#define _PATH(_LEV, X, Y)   [NSString stringWithFormat:@"%@/%@/%@_%@.%@", self.tileBasePath, @(_LEV), @(X), @(Y), self.tileFormat];
+    
+    if (level <= self.maxTileLevel) {
+        NSString *filename = _PATH(level, x, y);
+        return [UIImage imageWithContentsOfFile:filename];
+    }
+    
+    NSInteger gap = level - _maxTileLevel;
+    NSInteger glen = _tileSize >> gap;
+    NSInteger nm = 1<<gap;
+    NSInteger xoff=(x%nm)*glen, yoff=(y%nm)*glen;
+    
+    NSString *filename = _PATH(_maxTileLevel, (x>>gap), (y>>gap));
+    UIImage *oImg = [UIImage imageWithContentsOfFile:filename];
+    
+    CGImageRef imgRef = CGImageCreateWithImageInRect(oImg.CGImage, (CGRect){xoff, yoff, glen, glen});
+    oImg = [[UIImage alloc] initWithCGImage:imgRef];
+    CGImageRelease(imgRef);
+    
+    UIGraphicsBeginImageContextWithOptions(CGSizeMake(_tileSize, _tileSize), YES, 1.);
+    CGContextRef context = UIGraphicsGetCurrentContext();
+    CGContextSetInterpolationQuality(context, kCGInterpolationNone);
+    [oImg drawInRect:CGRectMake(0,0,_tileSize,_tileSize)];
+    UIImage *nImg = UIGraphicsGetImageFromCurrentImageContext();
```

Comment:
Minor, but why `oImg` and `nImg`? What's the `o` or `n` for? Maybe rename to full names such as `originalImage`, etc.?